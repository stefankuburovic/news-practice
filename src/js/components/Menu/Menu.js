import React from 'react';
import PropTypes from 'prop-types';

import MenuItem from '../MenuItem/MenuItem';
import './Menu.scss';

const Menu = ({
    items,
    position,
    setActive,
    active,
    isArticle,
    setActiveArticle,
    setActiveCategory,
}) => {
    return (
        <div className={`menu-container ${position}`}>
            {items.map((value, key) => {
                return (
                    <MenuItem
                        key={key}
                        item={value}
                        active={active}
                        isArticle={isArticle}
                        setActive={setActive}
                        setActiveArticle={setActiveArticle}
                        setActiveCategory={setActiveCategory}
                    />
                );
            })}
        </div>
    );
};

Menu.propTypes = {
    items: PropTypes.array,
    active: PropTypes.string,
    setActive: PropTypes.func,
    isArticle: PropTypes.bool,
    position: PropTypes.string,
    setActiveArticle: PropTypes.func,
    setActiveCategory: PropTypes.func,
};

export default Menu;
