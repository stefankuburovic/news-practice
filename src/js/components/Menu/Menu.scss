.menu-container {
  display: flex;
  flex-direction: row;

  &.right {
    margin-left: auto;
  }

  &.left {
    margin-right: auto;
  }
}