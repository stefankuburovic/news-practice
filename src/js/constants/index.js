/* eslint-disable react/jsx-key */
import { FormattedMessage } from 'react-intl';
import React from 'react';

export const MENU_ITEMS = [
    <FormattedMessage id="menu.topNews" defaultMessage="{text}" values={{ text: 'Top News' }} />,
    <FormattedMessage
        id="menu.categories"
        defaultMessage="{text}"
        values={{ text: 'Categories' }}
    />,
    <FormattedMessage id="menu.search" defaultMessage="{text}" values={{ text: 'Search' }} />,
];
export const LANGUAGES = [
    <FormattedMessage
        id="languageMenu.english"
        defaultMessage="{text}"
        values={{ text: 'English' }}
    />,
    <FormattedMessage
        id="languageMenu.german"
        defaultMessage="{text}"
        values={{ text: 'German' }}
    />,
];
export const COUNTRIES = ['GB', 'US'];
export const CATEGORIES = [
    'business',
    'entertainment',
    'general',
    'health',
    'science',
    'sports',
    'technology',
];
