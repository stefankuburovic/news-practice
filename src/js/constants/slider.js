export const SLIDER_SETTINGS = {
    dots: true,
    infinite: false,
    speed: 500,
    slidesToShow: 3,
    slidesToScroll: 3,
};
