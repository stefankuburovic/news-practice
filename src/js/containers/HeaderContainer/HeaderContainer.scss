@import '../../../scss/media_breakpoints';

.header {
  &-container {
    display: flex;
    position: relative;
    width: 100%;
    height: 50px;
    background: var(--red);
    @include media-breakpoint-down(md) {
      flex-direction: column;
      height: 100px;
    }
  }
}