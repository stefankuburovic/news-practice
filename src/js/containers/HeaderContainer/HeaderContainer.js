import React from 'react';
import PropTypes from 'prop-types';

import Menu from '../../components/Menu/Menu';
import { MENU_ITEMS, COUNTRIES } from '../../constants';
import LanguageSwitch from '../../components/LanguageSwitcher/LanguageSwitcher';
import './HeaderContainer.scss';

const HeaderContainer = ({
    active,
    country,
    isArticle,
    setActive,
    setActiveCountry,
    setActiveArticle,
    setActiveCategory,
}) => {
    return (
        <div className="header-container">
            <Menu
                position="left"
                active={active}
                items={MENU_ITEMS}
                setActive={setActive}
                setActiveArticle={setActiveArticle}
                setActiveCategory={setActiveCategory}
            />
            <LanguageSwitch />
            <Menu
                position="right"
                active={country}
                items={COUNTRIES}
                isArticle={isArticle}
                setActive={setActiveCountry}
                setActiveArticle={setActiveArticle}
            />
        </div>
    );
};
HeaderContainer.propTypes = {
    active: PropTypes.string,
    setActive: PropTypes.func,
    country: PropTypes.string,
    isArticle: PropTypes.bool,
    setActiveCountry: PropTypes.func,
    setActiveArticle: PropTypes.func,
    setActiveCategory: PropTypes.func,
};
export default HeaderContainer;
