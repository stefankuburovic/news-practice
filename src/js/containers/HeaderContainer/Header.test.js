import React from 'react';
import { shallow } from 'enzyme';
import HeaderContainer from './HeaderContainer';
import Menu from '../../components/Menu/Menu';
import { LANGUAGES, MENU_ITEMS } from '../../constants';

let wrapped = shallow(
    <HeaderContainer>
        <Menu items={MENU_ITEMS} position="left" />
        <Menu items={LANGUAGES} position="right" />
    </HeaderContainer>,
);

describe('HeaderContainer', () => {
    it('should render the HeaderContainer Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
});
