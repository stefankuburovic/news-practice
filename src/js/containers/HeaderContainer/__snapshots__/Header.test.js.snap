// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer should render the HeaderContainer Component correctly 1`] = `
<div
  className="header-container"
>
  <Menu
    items={
      Array [
        "Top News",
        "Categories",
        "Search",
      ]
    }
    position="left"
  />
  <Menu
    items={
      Array [
        "GB",
        "US",
      ]
    }
    position="right"
  />
</div>
`;
